### DISCUSSION TITLE
Import-Module

### DISCUSSION THREAD

**Discussion by WilliamTIII** [CrowdStrike/psfalcon#109] [Q&A] [answered]
I am unable to import the module. The error I receive is in the PSFalcon.psm1 file. the [Falcon] type is not defined and cannot be found. Also, despite moving the module to multiple different source directories, I am unable to import the module by name, I have to provide the full path. 
[PowerShell_transcript.DESKTOP-SMP17H4.P9n+IISs.20210218132336.txt](https://github.com/CrowdStrike/psfalcon/files/6005319/PowerShell_transcript.DESKTOP-SMP17H4.P9n%2BIISs.20210218132336.txt)

**Comment by bk-cs** [comment]
The error is due to the location that you unpacked the module, listed here:
C:\Windows\System32\WindowsPowerShell\v1.0\Modules\PSFalcon\psfalcon-master\PSFalcon.psm1

There are a few problems with this:
1. I've primarily tested PSFalcon in the typical "user modules" folder. It could behave differently if imported as a "system" or "all users" module.
2. It looks like that could potentially be referencing PowerShell v1.0 which--if that's the version actually being used to import the module--is not supported by PSFalcon. You'll need 5.1 or newer. I'm assuming your system has this, but it's possible that an older version is being utilized because of the module location.
3.  The `PSFalcon\psfalcon-master` structure will likely cause PowerShell to have trouble importing the files. It sounds like your error is due to the module not being able to find the `Class\Class.ps1` file that contains the `[Falcon]` type. You will never be able to import the module properly as long as that `psfalcon-master` folder is included in the path. PowerShell expects that the `PSFalcon.psd1` exists in the root of a folder called `PSFalcon`, or one below that, in a folder matching the module's version number (i.e. `PSFalcon\2.0.5`). I recommend using the version number as this is how the PowerShell Gallery does it, and I'm working to get the module on the PowerShell Gallery to help make future installation/updating easier.

Please try deleting everything PSFalcon-related, then following the [Installation instructions on the wiki](https://github.com/CrowdStrike/psfalcon/wiki/Installation) and let me know if that doesn't solve it.

**Comment by bk-cs** [comment]
Closing, pending further follow-up

**Comment by WilliamTIII** [comment, marked as answer]
I just got it to work - at least importing all of the commands. I had to
run PS as administrator, set my execution policy to unrestricted, use
unblock-file on the whole directory and I still was prompted to allow each
script to be ran. I think a digital signature would help. Thanks though,
your advice is appreciated.

**Comment by bk-cs** [comment]
I hear you on the signature and I'm working on that. Thanks for confirming!
